homeassistant/components/sensor/citybikes.py

```diff
@@ -250,13 +251,19 @@ def state(self):
         """Return the state of the sensor."""
         return self._station_data.get(ATTR_FREE_BIKES, STATE_UNKNOWN)
 
+    @property
+    def entity_id(self):
```

Comment:
Thanks for the input! My initial idea was to use `unique_id` (which seems to be the perfect field to store this kind of information). However, since the changes in the way entity IDs are applied, the code in Home Assistant prefers to use the `name` property for entity ID generation, as opposed to `unique_id` (i.e., if the `name` property isn't `None`, it will use it to generate `entity_id`, even if `unique_id` exists). See here: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_platform.py#L197

I feel that for this platform (it shows the number of bikes available at different bike-sharing stations), an entity ID built from the bike sharing network ID/name and the network-specific station ID is more appropriate than the name of the station (which is usually an address). Maybe I'm wrong and not up to date on the philosophy behind entity IDs. However, the address is definitely more appropriate as a friendly name.

As for your concern about duplicate entity IDs, this shouldn't be a problem, as the `entity_id` property is rewritten by HASS anyway, as you can see here: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_platform.py#L222